openerp/models.py

```diff
@@ -3394,8 +3394,8 @@ def qualify(field):
                     ))
             # store an access error exception in existing records
             exc = AccessError(
-                _('The requested operation cannot be completed due to security restrictions. Please contact your system administrator.\n\n(Document type: %s, Operation: %s)') % \
-                (self._name, 'read')
+                _('The requested operation cannot be completed due to security restrictions. Please contact your system administrator.\n\n(Document type: %s, Operation: %s, Records: %s, User: %s)') % \
+                (self._name, 'read', ','.join([str(r.id) for r in self]), self._uid)
```

Comment:
Records could be a large list. Maybe it's better limit this to, say, the first 6 records?